apps/price_pusher/src/controller.ts

```diff
@@ -75,7 +142,48 @@ export class Controller {
 
         // note that the priceIds are without leading "0x"
         const priceIds = pricesToPush.map((priceConfig) => priceConfig.id);
-        this.targetChainPricePusher.updatePriceFeed(priceIds, pubTimesToPush);
+
+        try {
+          // Start timers for each price update if metrics are enabled
+          const timers = this.metrics
+            ? pricesToPush.map((config) => ({
+                config,
+                timer: this.metrics!.startPriceUpdateTimer(
+                  config.id,
+                  config.alias,
+                ),
+              }))
+            : [];
+
+          await this.targetChainPricePusher.updatePriceFeed(
+            priceIds,
+            pubTimesToPush,
+          );
+
+          // Record successful updates and end timers
+          if (this.metrics) {
+            for (const { config, timer } of timers) {
+              this.metrics.recordPriceUpdate(config.id, config.alias);
+              timer(); // End the timer
+            }
+          }
+        } catch (error) {
+          this.logger.error(
+            { error, priceIds },
+            "Error pushing price updates to chain",
+          );
+
+          // Record errors in metrics
+          if (this.metrics) {
+            for (const config of pricesToPush) {
+              this.metrics.recordPriceUpdateError(
+                config.id,
+                config.alias,
+                error instanceof Error ? error.name : "unknown",
```

Comment:
i discourage this because you have no control over the name of the labels. let's just use the counter. later people can look through the logs.